pending/exports-and-boilerplate.pod

```diff
@@ -0,0 +1,116 @@
+=encoding utf8
+
+=head1 Reducing boilerplate and managing exports in Dancer2
+
+Dancer makes it easy to start small and grow fast. Today we will be looking at a techique to help manage that growth.
+
+A typical mature Dancer2 app will span multiple packages and each package will make use of Dancer's plugins and other modules available on CPAN.
+
+  use Scalar::Util qw< blessed >;
+  use List::Util qw< first all none sum >;
+  use Dancer2 appname => MyApp;
+  use Dancer2::Plugin::DBIC qw< schema resultset >;
+  use Dancer2::Plugin::Redis;
+
+After the first couple of files, the copy-pasting becomes tedious and if you later change your mind about what to include, you leave yourself open to runtime errors when you try to call a function you never imported into the package namespace.
+
+=head2 What Dancer2 "exports"
+
+Dancer2 "exports" almost eighty functions into the symbol table by default. That's a lot! You can get a complete list of them from the manual or from the symbol table itself with this one-liner:
+
+  perl -le 'use Dancer2; print for grep { main->can($_) && s/^\*main:://} map {$::{$_}} sort keys %main::;'
+
+(NB: If the concept of exporting and symbol tables in Perl is new to you, read C<perldoc -f use>. The documentation for L<Exporter> may also help you as we go along, but bear in mind that not all Perl modules - including Dancer2 - make use of Exporter.)
+
+For some applications this might present a problem: for instance, if you want to use L<List::Util/any> in a package where Dancer2 has been C<use>d, you cannot import it. Instead, you can qualify the C<any> from L<List::Util>:
+
+  any '/page/:id' => sub {
+    my $id = param('id');
+    forward '/401' if List::Util::any { $_ eq $id } @blocked_resources;
+    pass;
+  };
+
+If you're using the C<any> from List::Util more frequently than Dancer's C<any>, then you might decide you prefer to qualify the latter and import the former.
+
+However, you cannot selectively import functions from Dancer2 (this is something you might have noticed if you migrated from Dancer1 which allowed this). Nor will it help to C<require Dancer2>, as you will find that there is no such thing as C<Dancer2::any> (there was in Dancer1).
+
+The reason for this is the Dancer2's C<import> function (which is called when you C<use> it) does something clever, so that you to have multiple separate Dancer2 apps running in the same perl instance.
```

Comment:
You can link to the article in the Advent Calendar from 2014:
http://advent.perldancer.org/2014/10